drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_bellboard_local
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <haly/nrfy_bellboard.h>
+
+#define BELLBOARD_IRQ_COUNT 4U
```

Comment:
This number can potentially differ between cores in future devices. Perhaps it should be defined in devicetree.

Reply:
that's true, I guess there are 2 options here:

- Add `nordic,interrupt-lines = <N>;` in DT
- Always force `interrupts` to define the N lines, and add `nordic,interrupt-mask` which would indicate enabled lins (eg 0x3 = irq0 and irq1), this would require reworking the driver a bit, and would change how bellboard is used in DT, so it's a good time to decide.

The (2) looks more user-friendly at a first glance, but, AFAIK, the idea of using multiple IRQs is to play with priorities, in which case... one needs to re-define the `interrupts` property entirely. So (1) seems a better approach.